.github/CONTRIBUTING.md

```diff
@@ -1,8 +1,65 @@
-# How to Contribute
-
 We'd love to accept your patches and contributions to this project. There are
 just a few small guidelines you need to follow.
 
+## Repository Structure
+
+There are three main sections:
+
+**Codelabs** are interactive tutorials that are written in markdown syntax and published at [blocklycodelabs.dev](https://blocklycodelabs.dev). Codelabs mix natural language, code samples, and screenshots. The target user is following along and running the code as they read.
+
+The codelabs directory has a [template](https://github.com/google/blockly-samples/blob/master/codelabs/template.md) and one folder per codelab. Each codelab's folder contains the codelab (as a markdown file), and all associated assets (pngs, gif, etc).
+
+**Examples** are sample projects showing how to use Blockly features. Examples contain mostly code and a demo web page. Example code should be extremely well-commented to make it easy to copy. The target user may be reading the code, running it locally, or copying code snippets.
+
+The examples directory has one folder per example. Each example can be run with `npm install && npm run start`, and has a `README.md` file with additional context or instructions.
+
+**Plugins** are self-contained pieces of code that add functionality to Blockly. Plugins can add fields, define themes, create renderers, and much more. The target user is a developer who finds and uses the plugin through npm. Plugins defined in this repository are *first-party* plugins, which means that they are supported by the Blockly team.
+
+The plugins directory has one folder per plugin. To create the correct directory structure for your plugin, use [@blockly/create-package](https://github.com/google/blockly-samples/tree/master/plugins/dev-create#blocklycreate-package-).
+
+## How can I contribute?
+
+- Suggest plugins, examples, or codelabs
+- Implement plugins, examples, or codelabs
+- Report bugs
+- Fix bugs
+- Reproduce bugs
```

Comment:
I agree that all of those things are valuable--and for some of them, the reason I didn't include it is just that we don't have an answer to that question yet.  

I would like to punt on how to implement an example or a codelab into a separate PR so I can spend some more time writing and editing.

Bug reporting and feature requests--I'll links to the forms/

Specific tags--we don't have tags for new contributors or process around reproduction yet, but it's on my list.

Reply:
Sounds awesome! For some things, like codelabs or examples, it may be better to let people try it out first anyway. Then you can see where things are fine already, and where things need work.

Even if not all the detailed info is included in this first run, I think there's already enough here to get people started :D